From 1f6a003569cd8d068d56225a5ef840a2ea108cc6 Mon Sep 17 00:00:00 2001
From: Vignesh Viswanathan <quic_viswanat@quicinc.com>
Date: Tue, 1 Aug 2023 12:12:50 +0530
Subject: [PATCH] remoteproc: qcom: Trigger BUG_ON from SSR notify crash
 handler

Trigger BUG_ON from SSR notify crash handler as a temporary WAR as SSR
notify crash notification is not received for MPD crashes in the client
driver.

Change-Id: I8cef12c98d2db15654b802c3b7078eaed227ddc8
Signed-off-by: Vignesh Viswanathan <quic_viswanat@quicinc.com>
---
 drivers/remoteproc/qcom_common.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/remoteproc/qcom_common.c b/drivers/remoteproc/qcom_common.c
index 24270b9e73dc..1aaeb1419506 100644
--- a/drivers/remoteproc/qcom_common.c
+++ b/drivers/remoteproc/qcom_common.c
@@ -519,6 +519,8 @@ static void ssr_notify_crash(struct rproc_subdev *subdev)
 
 	atomic_notifier_call_chain(&ssr->info->atomic_notifier_list,
 				 QCOM_SSR_NOTIFY_CRASH, &data);
+
+	BUG_ON(1);
 }
 
 /**
-- 
2.34.1

